src/artm/core/dictionary.cc

```diff
@@ -43,11 +43,13 @@ Dictionary::Dictionary(const artm::DictionaryConfig& config) {
         }
 
         // std::map::insert() ignores attempts to write multiply pairs with same key
-        // the data representation is symmetric for both first and second tokens in cooc pair
         first_cooc_iter->second.insert(std::make_pair(second_index_iter->second,
                                                       config.cooc_entries().value(i)));
-        second_cooc_iter->second.insert(std::make_pair(first_index_iter->second,
-                                                       config.cooc_entries().value(i)));
+
+        if (config.cooc_entries().symmetric_cooc_values()) {
+          second_cooc_iter->second.insert(std::make_pair(first_index_iter->second,
+                                                         config.cooc_entries().value(i)));
+        }
```

Comment:
Can we place strings 39-43 also under if(config.cooc_entries().symmetric_cooc_values())? 
To create second_cooc_iter only if we are going to use it.